tokens/token-2022/anchor/tests/anchor.ts

```diff
@@ -0,0 +1,179 @@
+import * as anchor from "@coral-xyz/anchor";
+import { Program } from "@coral-xyz/anchor";
+import { Anchor } from "../target/types/anchor";
+
+const loadKeypairFromFile = (path: string): anchor.web3.Keypair => {
+  return anchor.web3.Keypair.fromSecretKey(
+    Buffer.from(JSON.parse(require("fs").readFileSync(path, "utf-8")))
+  );
+};
+
+describe("anchor", () => {
+  // Configure the client to use the local cluster.
+  anchor.setProvider(anchor.AnchorProvider.env());
+
+  const program = anchor.workspace.Anchor as Program<Anchor>;
+  const connection = program.provider.connection;
+  const TOKEN_2022_PROGRAM_ID = new anchor.web3.PublicKey(
+    "TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb"
+  );
+  const payer = anchor.web3.Keypair.generate();
+  const ATA_PROGRAM_ID = new anchor.web3.PublicKey(
+    "ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL"
+  );
+
+  const tokenName = "TestToken";
+  const [mint] = anchor.web3.PublicKey.findProgramAddressSync(
+    [
+      Buffer.from("token-2022-token"),
+      payer.publicKey.toBytes(),
+      Buffer.from(tokenName),
+    ],
+    program.programId
+  );
+  const [payerATA] = anchor.web3.PublicKey.findProgramAddressSync(
+    [
+      payer.publicKey.toBytes(),
+      TOKEN_2022_PROGRAM_ID.toBytes(),
+      mint.toBytes(),
+    ],
+    ATA_PROGRAM_ID
+  );
+
+  const receiver = anchor.web3.Keypair.generate();
+
+  connection.requestAirdrop(receiver.publicKey, 1000000000);
+
+  const [receiverATA] = anchor.web3.PublicKey.findProgramAddressSync(
+    [
+      receiver.publicKey.toBytes(),
+      TOKEN_2022_PROGRAM_ID.toBytes(),
+      mint.toBytes(),
+    ],
+    ATA_PROGRAM_ID
+  );
+
+  it("Create Token-2022 Token", async () => {
+    const tx = new anchor.web3.Transaction();
+
+    const ix = await program.methods
+      .createToken(tokenName)
+      .accounts({
+        mint: mint,
+        signer: payer.publicKey,
+        tokenProgram: TOKEN_2022_PROGRAM_ID,
+      })
+      .instruction();
+
+    tx.add(ix);
+
+    const sig = await anchor.web3.sendAndConfirmTransaction(
+      program.provider.connection,
+      tx,
+      [payer]
+    );
+    console.log("Your transaction signature", sig);
+  });
+
+  it("Initialize payer ATA", async () => {
+    const tx = new anchor.web3.Transaction();
+
+    const ix = await program.methods
+      .createAssociatedTokenAccount()
+      .accounts({
+        tokenAccount: payerATA,
+        mint: mint,
+        signer: payer.publicKey,
+        tokenProgram: TOKEN_2022_PROGRAM_ID,
+      })
+      .instruction();
+
+    tx.add(ix);
+
+    const sig = await anchor.web3.sendAndConfirmTransaction(
+      program.provider.connection,
+      tx,
+      [payer]
+    );
+    console.log("Your transaction signature", sig);
+  });
+
+  /*
+  // Comment out because we use init in the transfer instruction
```

Comment:
As an example reference, yes. The payer's ATA in the tests are also created with this instruction.

Reply:
Is the process for creating an ATA for a Token2022 mint different than legacy mints?

Reply:
nvm, I see where the reference is needed. 
Yeah, good with me, leave it in. Just put a little message in your comment header that mentions that you're including this instruction as a reference, but it's not required for the test. 
The message you have there isn't super clear at first why it's commented out

Reply:
Something like "This instruction is included only as a reference, but is not required to run this test, because we are using `init` in the transfer instruction. The `create_associated_token_account` instruction on the program is provided as a reference as well"

Reply:
Changes made, ready to merge.